### ISSUE TITLE
Synchronous attributeChangedCallbacks

### ISSUE THREAD

**Issue by opinel** [WebReflection/document-register-element#16]
Hi again,

coming from the same perspective as in my previous pull-request I was wondering how we could improve  the polyfill experience of the `attributeChangeCallbacks` compared to the native experience?
A native attribute change would trigger an event callback in the same run-loop. This would also happen for the polyfill implementation of IE9. However, in Safari or the current Firefox the attribute change event would only be noticed in the next MutationObserver callback.
What about changing the polyfill to always behave synchronous as in the case of IE9? What about making the handling of these changes faster by returning earlier on `style` attribute changes?

**Comment by WebReflection** [comment]
the TL;DR answer is: it's a feature, not a bug.

The reason `MutationObserver` is used is because it's faster and it scales more when you change many things at once. It's the `requestAnimationChange` equivalent for attribute changes and I believe even the native implementation works in a similar way behind the scene.

You don't want to change in a blocking/synchronous way attributes because that won't perform any good for bigger projects. Each little attribute change might trigger synchronous behaviors in all subnodes of a custom elements too or vice versa.

It's like changing the window size on resize event, you don't know when that blocking behavior would stop. Having that batched and optimized natively is a great thing.

Now, I understand the createdCallback gotcha and I've provided a work around for that, but I start believing you really have some problem with asunchronous code in general ... or if you could be so kind to explain me why would you want or need that, I might re-consider this answer.

**Comment by WebReflection** [comment]
P.S. `MutationObserver` like `Object.observe` were introduced to exactly avoid synchronous blocking changes, even to regular Objects. If I were you, I rather try to figure out how to deal with these scenarios instead of asking everyone and everything to be sync and blocking. Just saying, those APIs are not there to annoy developers but to actually help them.

**Comment by opinel** [comment]
TL;DR: I just want to have a real consistent cross-browser behavior. 
If I would override the `setAttribute` method as you do in one of your examples, I would get this behavior out of the box. The only use-case I would not support would be the manipulation through the web-inspector in the developer's browser.

**Comment by WebReflection** [comment]
You have a consistent behavior, you define what should happen in the CustomElement prototype and events definition and you got it. **That is** what this polyfill is about, an API that works across platform consistently and **it does** ... so that's the good news.

The bad one is, you are apparently doing things that won't even work on Polymer since it uses same `MutationObserver` behavior because that's what you want to do as well but you don't know it yet ;-)

**Comment by WebReflection** [comment]
btw, you didn't give me a single vald use case for this request, so I am going to close this. Feel free to change my mind with real needs and please do not mention your testing frameworks, thank you for your understanding.
